### DISCUSSION TITLE
HTTP scaling URL-based

### DISCUSSION THREAD

**Discussion by lgmorand** [kedacore/keda#1620] [Q&A / Need Help]
Hi guys

I'd like to use keda + an "http trigger". Basically, I found that I could use prometheus scaler scanning for nginx metrics but the example I found is based on http-total-requests

I have:
- 1 ingress controller
- 3 deployments
- need to scale them with different scaleobject/config

what I'd like is to scale depending on http-total-request for a specific URL (= per ingress route). I'm not sure that nginx exposes these detailed metrics

An alternative could be to use azure monitor if all requests were logged but there is a delay between logging and the ability to query it.

What would be your recommandation for this kind of usecase ?

thanks
@tomkerkhove  => sorry to ping you directly but I trust you blindly :)

**Comment by tomkerkhove** [comment]
I'm not an expert on the nginx metrics so will not be able to help I'm afraid, but if they provide it; you can scale on it.

We also have an *experimental* project to support this out-of-the-box which is available https://github.com/kedacore/http-add-on

**Comment by lgmorand** [comment]
Thanks. I digged a little bit.

Nginx can expose these metrics per "url" IF we define "zone location" AND we use a specific module AND we use Nginx+ instead of Nginx

An alternative I'm gonna try is to test with linkerd which can expose metrics per URL with its service profiles (it seems).

I'll try to check the new addon. I found it on the blog (https://hackmd.io/@arschles/kedahttp) but seemed to have too many components while my customer is not super mature yet. 

thanks for your feedback, I'll come back if we have a fully working implementation

**Reply by tomkerkhove** [reply]
Thanks! In terms of KEDA HTTP, what would make it easier for you to use?

We can see if we can make it simpler.
